### PR TITLE
Fast ReLU using bitwise operators

### PR DESCRIPTION
**Pull request by ziachap**
Here's an implementation of ReLU that I found to be around 4x faster than the default ReLU in my own benchmarks.

### CODE REVIEW

File: src/SharpNeat/NeuralNets/Double/ActivationFunctions/BitwiseReLU.cs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+/// Rectified linear activation unit (ReLU) using bitwise operators.
+/// </summary>
+public sealed class BitwiseReLU : IActivationFunction<double>
+{
+    /// <inheritdoc/>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    public void Fn(ref double x)
+    {
+        long sign = BitConverter.DoubleToInt64Bits(x) >> 63;
+        x *= (sign & (-sign >> 63)) | (~sign & 1);
```
Comment (colgreen) [inline src/SharpNeat/NeuralNets/Double/ActivationFunctions/BitwiseReLU.cs:18]: I think `(sign & (-sign >> 63))` always evaluate to zero and is therefore redundant(?)

Comment (ziachap) [reply src/SharpNeat/NeuralNets/Double/ActivationFunctions/BitwiseReLU.cs]: You're right, I've removed that section and its still passing my tests.

Comment (colgreen) [reply src/SharpNeat/NeuralNets/Double/ActivationFunctions/BitwiseReLU.cs]: Here's an even faster version that avoids the multiplication:

```
    private static void BitwiseFn2(ref double x)
    {
        long xlong = BitConverter.DoubleToInt64Bits(x);
        x = BitConverter.Int64BitsToDouble(xlong & ~(xlong >> 63));
    }
```

For JITter output see:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgDkBXfGKASzFwBuGjWIBmJqQYBhBgG8aDJQ0XKADnwBu2DDCbkkTFAwBCvDAHdeuGADEAdgApYAMwYATCB2AAbPQgBKVSUFamVwhh8IewBzBlxeGPsGAF5Tc2lozR5dKBYAES9fGAAVCABJewxUMwxcR0CGAD4mhiQxYTCIpQQGACo0xwA/BKSGADIGcgDO8IBfcOCVLqUNXm1dfUNiY1qrGwdSZxg3T28/BkCl0O6lKNjL+7i02sz7bKhcgqK/MsrqlC1eqBWa3XovDJZHI8Fj/GrmXBlQrnGANJ4TBhDNHROItNpiAIzJZzEQrBhrDZ6ZjbYwOY6nH7+IJkm63XguBoMAA8DDoLDozNuEXBvP5oKUJOocyAA===

**Comment by colgreen** [comment]
Thanks for the PR. It's an interesting approach that can probably be used in multiple other places.

**Pull request merged**